### PR TITLE
cts test_context support

### PR DESCRIPTION
**Pull request by airlied**
This builds on the previous pull request for select_device and allows the CTS test_context to build, run and pass here.

### CODE REVIEW

**Comment by keryell** [comment]
I guess now you can rebase this onto master.

**Comment by airlied** [comment]
I've rebased this and I've fixed one size->empty and wrong name in new file.

**Comment by airlied** [comment]
I'll look into the CI fails next week sometime.

**Comment by airlied** [comment]
@keryell so it appears the cts failure are that we call a sycl runtime error when we can't find a device now with the passed in selector, I think that is correct according to the spec. should we fix the tests to handle this, or have some of the tests fall back to cpu execution?

**Comment by keryell** [comment]
That is true.
Probably we can rewrite the example to request a context & queue with a GPU inside a `try`/`catch` and if it fails just use the host version.
The context is not useful here anyway, even if it was on some slides 4 years ago... :-)

File: tests/2014-04-21-HPC-GPU_Meetup/slide_13.cpp [hunk 1]
```diff
@@ -34,6 +39,7 @@ int main() {
                                                   }
                                                   );
           });
+        delete myQueue;
```
Comment (keryell) [inline tests/2014-04-21-HPC-GPU_Meetup/slide_13.cpp]: Can be removed if no pointer anymore.

**Comment by airlied** [comment]
Any more needed for this to land?

**Comment by keryell** [comment]
That is very curious: your last commit https://github.com/triSYCL/triSYCL/tree/9a19b15f2d7d539c4e22428aa0437e8731a0020c does not appear in your `cts-context` branch...
I think I made some comments on it recently but it does not appear either...
 
How is it possible?

**Pull request closed without merging**